engines/sword1/text.cpp

```diff
@@ -99,9 +101,18 @@ void Text::makeTextSprite(uint8 slot, const uint8 *text, uint16 maxWidth, uint8
 	memset(linePtr, NO_COL, sprSize);
 	for (lineCnt = 0; lineCnt < numLines; lineCnt++) {
 		uint8 *sprPtr = linePtr + (sprWidth - lines[lineCnt].width) / 2; // center the text
+		Common::String textString;
+		const uint8 *curTextLine = text;
+		if (SwordEngine::_systemVars.isLangRtl) {
+			Common::String textLogical = Common::String((const char *)text, (uint32)lines[lineCnt].length);
+			textString = Common::convertBiDiString(textLogical, Common::kWindows1255);
+			curTextLine = (const uint8 *)textString.c_str();
+		}
 		for (uint16 pos = 0; pos < lines[lineCnt].length; pos++)
-			sprPtr += copyChar(*text++, sprPtr, sprWidth, pen) - OVERLAP;
-		text++; // skip space at the end of the line
+			sprPtr += copyChar(*curTextLine++, sprPtr, sprWidth, pen) - OVERLAP;
+		curTextLine++; // skip space at the end of the line
+		text += lines[lineCnt].length + 1;
```

Comment:
I'm not sure I understand the question.
Do you mean in line 114, why isn't it

		curTextLine+= lines[lineCnt].length + 1;
?

Well, if that's the question:
1. It simply doesn't work with (Hebrew or English) text that has more than one line (I've verified it now)
2. When we have a Hebrew text, `text` contains the original, unmodified, block of text. Each iteration it advances to the beginning of a new line in the original text. On the other hand, `curTextLine` contains a pointer to the beginning of a new *reversed* line. Thus, we want to take in each iteration the pointer to the 'clean', unmodified `text` line, to avoid leftovers and other surprises.

Reply:
That indeed looks correct.
The old code is advancing the `text` pointer inside the for loop. But as this is no longer the case, this needs to be done here so that the next `lineCnt` iteration has the correct `test` pointer.